### ISSUE TITLE
How to not automatically convert header keys to lower case?

### ISSUE THREAD

**Issue by ngugc** [sagebind/isahc#205]
The servers which expect header keys to be case sensitive are broken.
Is there any way to ignore the conversion?

```
    let client = HttpClient::builder()
        .timeout(Duration::from_secs(5))
        .redirect_policy(RedirectPolicy::Follow)
        .build()?;

    let req = Request::get("https://rust-lang.org")
        .header("Key", "Hello")
        .body(Body::empty())?;
    println!("Req: {:?}", req);

```
```
Req: Request { method: GET, uri: https://rust-lang.org/, version: HTTP/1.1, headers: {"key": "Hello"}, body: Body(0) }
```

**Comment by sagebind** [comment]
Thanks for your question! I don't think there's a way to force a specific header name casing, it is being coerced into lowercase internally it seems by the lower-level [http](https://docs.rs/http) library. Here's the note about this: https://docs.rs/http/0.2.1/http/header/struct.HeaderName.html#representation

I'm not sure if there's a workaround for this or if I would be willing to implement such a workaround, since being case-sensitive is against the HTTP spec. I'll have to think more about this one.

**Comment by sagebind** [comment]
After mulling it over, I think we can support something minimal for HTTP/1.x. Note that in HTTP/2 and newer, headers are _always_ lowercase.

While we _can't_ preserve the header case, we _can_ send headers using `Title-Case` under HTTP/1.x as a new configuration option. @ngugc Would the option to use title-case solve your problem? Basically:

- `Foo` is stored as `foo`, we send it as `Foo`
- `FOOBar` is stored as `foobar`, we send it as `Foobar`
- `foo-bar` is stored as `foo-bar`, we send it as `Foo-Bar`
-  etc

This is basically copying a page from reqwest's book, which has an option named [`http1_title_case_headers`](https://docs.rs/reqwest/0.10.7/reqwest/struct.ClientBuilder.html#method.http1_title_case_headers).

**Comment by ngugc** [comment]
I think at least it will fix my current problem.
Thanks!

**Comment by sagebind** [comment]
This is now available in the 0.9.8 release as [`Configurable::title_case_headers`](https://docs.rs/isahc/0.9.8/isahc/config/trait.Configurable.html#method.title_case_headers)!
